### PR TITLE
feat: fuel account connection

### PR DESCRIPTION
**Pull request by matt-user**
closes FRO-6

### CODE REVIEW

**Comment by matt-user** [comment]
making this pr a draft until https://github.com/FuelLabs/fuels-wallet/pull/726 is merged and I can use the fix

File: packages/hooks/src/hooks/useAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    async () => {
+      if (!fuel) {
+        return null;
+      }
+      try {
+        const currentFuelAccount = await fuel.currentAccount();
+        return currentFuelAccount;
+      } catch (error: unknown) {
+        return null;
+      }
```
Comment (LuizAsFight) [inline packages/hooks/src/hooks/useAccount.tsx:40]: let's not use `null`. also this code can be smaller

```suggestion
      try {
        const currentFuelAccount = await fuel?.currentAccount();
        return currentFuelAccount;
      } catch (e) {
        return undefined;
      }
```

File: packages/hooks/src/hooks/useConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  // TODO: wrap in useCallback if we add more functionality/args
+  // to the mutationFn
+  const connect = mutate;
+
+  // TODO: implement a connectAsync
+
+  return {
+    connect,
+    data,
```
Comment (LuizAsFight) [inline packages/hooks/src/hooks/useConnect.tsx:35]: can we remove the comments?

```suggestion
  return {
    connect: mutate,
    data,
```

File: packages/app/src/systems/Accounts/hooks/useAccountConnectionFuel.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const { disconnect } = useDisconnect();
+
+  return {
+    handlers: {
+      connect,
+      disconnect,
+      openFuelInstall: store.openFuelInstall,
+      closeDialog: store.closeOverlay,
+    },
+    fuel,
```
Comment (LuizAsFight) [inline packages/app/src/systems/Accounts/hooks/useAccountConnectionFuel.tsx:23]: we should not return `fuel`, could return something like:
`hasInstalledFuel: Boolean(fuel)`

File: packages/hooks/src/hooks/useAccount.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    },
+    {
+      enabled: !!fuel,
+    }
+  );
+
+  return {
+    address: data,
+    isConnected: !!data,
```
Comment (LuizAsFight) [inline packages/hooks/src/hooks/useAccount.tsx:49]: I think this line is a problem as we can have different results from `fuel.isConnected` function. `useAccount` hook should only return `address`.. let's keep `isConnected` to other hook

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
1- let's include new hooks that will be needed really soon:
`useIsConnected`
`useProvider`

2- let's make all hooks have the same output pattern:
consider that `queryProps` is the output of both `useMutation`/`useQuery`. like onLoading, isIdle, etc...

output of hooks will be:
`{ data, handlers, ...queryProps }`

for example `useAccount`
```
return {
  data: address,
  ...queryProps
}
```


for example useConnect
```
return {
  data,
  handlers: {
    connect
  },
  ...queryProps
}
```


on this way we know that for every hook `data` will be inside data props. actions will be inside `handlers` prop, and all other props are related to query status/loading/errors/etc

File: packages/app/src/systems/Accounts/components/FuelInstallPopup/FuelInstallPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  );
+};
+
+const styles = {
+  description: cssObj({
+    display: 'flex',
+    justifyContent: 'center',
+  }),
+  heading: cssObj({
+    color: 'Black',
```
[the remark this answers is not in the document]
Comment (matt-user) [reply packages/app/src/systems/Accounts/components/FuelInstallPopup/FuelInstallPopup.tsx:40]: yes, but this pr does not have the new fuel-ui version.  That pr is what I will work on next and will fix this issue.

File: packages/app/src/systems/Accounts/pages/AccountConnectionFuel.tsx [hunk 6]
```diff
@@ -3,9 +3,9 @@ import { AccountConnectionInput } from '../components';
 import { useAccountConnectionFuel } from '../hooks';
 
 export const AccountConnectionFuel = () => {
-  const { isLoading, handlers, hasInstalledFuel, address } =
+  const { isLoading, handlers, hasInstalledFuel, account } =
     useAccountConnectionFuel();
-  const nonNullAddress = !address ? undefined : address!;
+  const nonNullAddress = !account ? undefined : account!;
```
Comment (LuizAsFight) [inline packages/app/src/systems/Accounts/pages/AccountConnectionFuel.tsx]: this line is not needed anymore

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: packages/sdk-react/src/hooks/useAccount.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      try {
+        const currentFuelAccount = await fuel.currentAccount();
+        return currentFuelAccount;
+      } catch (error: unknown) {
+        return null;
+      }
+    },
+    {
+      enabled: !!fuel,
+    }
```
Comment (LuizAsFight) [inline packages/sdk-react/src/hooks/useAccount.tsx:23]:
```suggestion
      try {
        const currentFuelAccount = await fuel?.currentAccount();
        return currentFuelAccount || null;
      } catch (error: unknown) {
        return null;
      }
    }
```

File: packages/sdk-react/src/hooks/useIsConnected.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { IS_CONNECTED_KEY, useFuelReactContext } from '../components';
+
+export const useIsConnected = () => {
+  const { fuel } = useFuelReactContext();
+
+  const { data, ...queryProps } = useQuery([IS_CONNECTED_KEY], async () => {
+    const isConnected = await fuel?.isConnected();
+    return isConnected;
```
Comment (LuizAsFight) [inline packages/sdk-react/src/hooks/useIsConnected.tsx:10]:
```suggestion
    return isConnected || false;
```

File: packages/sdk-react/src/hooks/useIsConnected.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  });
+
+  return {
+    isConnected: data || undefined,
```
Comment (LuizAsFight) [inline packages/sdk-react/src/hooks/useIsConnected.tsx:14]:
```suggestion
    isConnected: data,
```

File: packages/sdk-react/src/hooks/useProvider.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { PROVIDER_KEY, useFuelReactContext } from '../components';
+
+export const useProvider = () => {
+  const { fuel } = useFuelReactContext();
+
+  const { data, ...queryProps } = useQuery([PROVIDER_KEY], async () => {
+    const provider = await fuel?.getProvider();
+    return provider;
```
Comment (LuizAsFight) [inline packages/sdk-react/src/hooks/useProvider.tsx:10]:
```suggestion
    const provider = await fuel?.getProvider();
    return provider || null;
```

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Review by LuizAsFight — APPROVED** [review]
as @matt-user will be OOO for some days I will approve this PR as-is and implement remaining (minor) requested changes in this PR #38

**Pull request merged**